File: erp-backend/xingyun-sc/src/main/java/com/lframework/xingyun/sc/vo/stock/serial/QueryProductStockSerialVo.java
package com.lframework.xingyun.sc.vo.stock.serial;

import com.lframework.starter.web.core.vo.PageVo;
import io.swagger.annotations.ApiModelProperty;
import lombok.Data;

/**
 * 查询序列号库存的参数
 */
@Data
public class QueryProductStockSerialVo extends PageVo {

    private static final long serialVersionUID = 1L;

    /**
     * 仓库ID
     */
    @ApiModelProperty(value = "仓库ID")
    private String scId;

    /**
     * 商品编号
     */
    @ApiModelProperty(value = "商品编号")
    private String productCode;

    /**
     * 商品名称
     */
    @ApiModelProperty(value = "商品名称")
    private String productName;

    /**
     * 商品类目ID
     */
    @ApiModelProperty(value = "商品类目ID")
    private String categoryId;

    /**
     * 商品品牌ID
     */
    @ApiModelProperty(value = "商品品牌ID")
    private String brandId;

    /**
     * 序列号
     */
    @ApiModelProperty(value = "序列号")
    private String serialNumber;
}

File: erp-frontend/src/api/sc/stock/product-stock-serial/model/queryProductStockSerialVo.ts
import { PageVo } from '@/api/model/pageVo';

/**
 * 查询航材序列号库存请求参数
 */
export interface QueryProductStockSerialVo extends PageVo {
  // 仓库ID
  scId?: string;

  // 航材编号
  productCode?: string;

  // 航材名称
  productName?: string;

  // 航材分类ID
  categoryId?: string;

  // 航材制造商ID
  brandId?: string;
  
  // 序列号
  serialNumber?: string;
}
